A continuación se muestra una matriz de expresión genética:

In [1]:
genexp<-read.csv("practica/GEMatrixFiltered.csv")
head(genexp)

X              X212284_x_at X212869_x_at X213614_x_at X212391_x_at
1 Normal.BC0043N  9167.03      7380.27     8341.82       7944.99    
2 Cancer.BC0043T 10635.90     11010.80     9859.02      10493.60    
3 Normal.BC0051N  8015.29      6112.62     8287.92       7360.60    
4 Cancer.BC0051T  9081.20      8925.35     8276.54       9021.02    
5 Normal.BC0057N  9006.52      8338.44     7069.53       7909.77    
6 Cancer.BC0057T  8307.87      7295.69     7048.75       7111.87    
  X201257_x_at AFFX.hum_alu_at X206559_x_at X214003_x_at X208755_x_at ⋯
1  8367.28     7884.22         6209.89      6009.60      9017.07      ⋯
2 10805.70     8507.10         9060.41      9426.21      8674.11      ⋯
3  7060.93     8003.40         7140.51      6029.11      7026.17      ⋯
4  8622.68     8925.17         7862.39      6847.31      9651.01      ⋯
5  7348.80     7966.06         7120.25      7903.76      7387.18      ⋯
6  6372.62     8056.31         6406.97      6256.54      7098.11      ⋯
  X214981_at X216660_at X217101_at X218922_s_at X203953_s_at X221755_at
1 13.9187    30.9902    26.8305    30.6226      13.6677      23.7455   
2 17.2673    56.3875    16.9442    14.5954      15.2555      19.2764   
3 26.6429    29.1446    23.8505    32.2064      15.9346      27.4183   
4 37.5716    30.0083    20.7826    22.1020      27.5567      26.4899   
5 36.4167    29.9436    45.3716    31.1614      31.3397      32.0059   
6 21.3444    54.1291    37.4840    22.2405      50.9866      26.8458   
  X208278_s_at X207162_s_at X206824_at X218840_s_at
1 29.1401      23.8904      64.3171    25.5825     
2 16.4308      26.5672      53.7115    28.7201     
3 21.6383      21.3962      27.9609    45.1721     
4 26.3171      16.6457      41.5037    25.8667     
5 29.1746      21.2647      34.7236    35.4706     
6 33.2448      18.3316      24.6310    24.8679

In [7]:
ncol(genexp)
nrow(genexp)


[1] 482

[1] 86

### Limpieza de datos:

* Normalizar
* ¿Existen outliers? Tratamiento de Outliers

### Parte 1
Realizar un modelo de regresión logistica que sea capaz de, dada la expresión genética de 482 genes realice una predicción de si el tejido es canceroso o no:
* Pintar curva ROC
* Para un umbral de 0 calcular:
 * Precisión
 * F0.5
 * Exactitud
 * Exhaustividad
 * Matriz de confusión
 

### Parte 2

Realizar clusterización por k-means y uno basado en áboles (dendrograma). ¿Cual sería el número de clusters óptimo?  ¿Puedes identificar algún cluster interesante?

### Parte 3

Realizar un PCA, quedarse con los dos valores más representativos. Pintarlo en una nube de puntos donde los tejidos normales tendrán un color y los tejidos con cancer otro.

A partir de esta representación simplificada (con solo 2 valores más significativos) crear un modelo de regresión logística. Compara la curva ROC de este nuevo modelo con el de la *Parte 1*